pkg/apis/machine/v1alpha1/machineclass_types.go

```diff
@@ -39,18 +40,17 @@ type MachineClass struct {
 	// +optional
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
+	// +kubebuilder:validation:XPreserveUnknownFields
+	// AutoscalerNodeTemplate contains subfields to track all node resources required to scale nodegroup from zero
+	AutoscalerNodeTemplate AutoscalerNodeTemplate `json:"autoscalerNodeTemplate"`
```

Comment:
we might have to make this field `optional` (and so pointer as well) for backward compatibility as stated for the `NodeTemplate` field by Tim [here](https://github.com/gardener/gardener/pull/4980#discussion_r754852941)

Reply:
Yes, I had already made a point about it.